cl_dll/hud.cpp

```diff
@@ -301,6 +301,42 @@ void __CmdFunc_Append()
 	EngineClientCmd(gEngfuncs.Cmd_Argv(1));
 }
 
+void __CmdFunc_Savemap()
+{
+	FILE* saved_maps;
+	char map_name[64];
+	char map_name_to_check[64];
+
+	get_map_name(map_name, ARRAYSIZE(map_name));
+
+	if (map_name[0])
+	{
+		saved_maps = fopen("saved_maps.txt", "r+"); // returns NULL if the file doesn't exist or if the file is read-only
+
+		if (saved_maps)
+		{
+			while (fgets(map_name_to_check, ARRAYSIZE(map_name_to_check), saved_maps))
+			{
+				map_name_to_check[strlen(map_name_to_check) - 1] = '\0';
+
+				if (!strcmp(map_name, map_name_to_check))
+				{
+					gEngfuncs.Con_Printf("Current map is already in saved_maps.txt\n");
+					fclose(saved_maps);
+					return;
+				}
+			}
+		}
+		else if(!(saved_maps = fopen("saved_maps.txt", "a"))) // checks if the folder is read-only. If yes - saved_maps is NULL, if not - creates the file
```

Comment:
Uhhh, sorry for nitpicking but could you please not use assignment inside an if condition? It frequently leads to hard to notice bugs. Other than that this PR is fine.

```c++
else
{
    saved_maps = fopen("saved_maps.txt", "a");
    if (!saved_maps)
        return;
}
```
should be sufficient.